django/forms/models.py

```diff
@@ -49,7 +49,8 @@ def construct_instance(form, instance, fields=None, exclude=None):
         # Leave defaults for fields that aren't in POST data, except for
         # checkbox inputs because they don't appear in POST data if not checked.
         if (f.has_default() and
-                form[f.name].field.widget.value_omitted_from_data(form.data, form.files, form.add_prefix(f.name))):
+                form[f.name].field.widget.value_omitted_from_data(form.data, form.files, form.add_prefix(f.name)) and
+                cleaned_data.get(f.name) in (None, '')):
```

Comment:
IMO we should re-use existing `form[f.name].field.empty_values` instead of `(None, '')`.

Reply:
Wasn't aware of this attribute. Thanks.